modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/ServerApiAccess.java

```diff
@@ -0,0 +1,327 @@
+		Iterator<Entry<String, String>> iterator = params.entrySet().iterator();
+		while (iterator.hasNext()) {
+			Entry<String, String> param = iterator.next();
+			bodyBuilder.append(param.getKey()).append('=').append(param.getValue());
+			if (iterator.hasNext()) {
+				bodyBuilder.append('&');
+			}
+		}
+		return bodyBuilder.toString();
+	}
+
+	public static HttpRequestBase buildHeaders(HttpRequestBase httpRequestBase,
+	                                           Map<String, String> headers, String httpMethod) {
+		Iterator<Entry<String, String>> iterator = headers.entrySet().iterator();
+		while (iterator.hasNext()) {
+			Entry<String, String> header = iterator.next();
+			httpRequestBase.setHeader(header.getKey(), header.getValue());
+		}
+		return httpRequestBase;
+	}
+
+	public static Map<String, String> postData(APIUtilities apiUtilities,
+	                                           Map<String, String> headers) {
+		String httpMethod = apiUtilities.getHttpMethod();
+		String url = apiUtilities.getEndPoint();
+		JSONObject params = apiUtilities.getRequestParams();
+		Map<String, String> responseParams = new HashMap<String, String>();
+		HttpClient httpclient = getCertifiedHttpClient();
+
+		if (httpMethod.equals("POST")) {
+			HttpPost httpPost = new HttpPost(url);
+			if (params != null) {
+				try {
+					httpPost.setEntity(new StringEntity(params.toString()));
+				} catch (UnsupportedEncodingException e) {
+					e.printStackTrace();
+				}
+			} else {
+				httpPost.setEntity(null);
+			}
+			Log.e("url", "" + url);
+			HttpPost httpPostWithHeaders = (HttpPost) buildHeaders(httpPost, headers, httpMethod);
+			try {
+				HttpResponse response = httpclient.execute(httpPostWithHeaders);
+				String status = String.valueOf(response.getStatusLine().getStatusCode());
+				Log.d(TAG, status);
+				responseParams.put("response", getResponseBody(response));
+				responseParams.put("status", status);
+				return responseParams;
+			} catch (ClientProtocolException e) {
+				Log.d(TAG, "ClientProtocolException :" + e.toString());
+				return null;
+			} catch (IOException e) {
+				Log.d(TAG, e.toString());
+				responseParams.put("response", "Internal Server Error");
+				responseParams.put("status", "500");
+				return responseParams;
+			}
+		} else if (httpMethod.equals("GET")) {
+			// if(payload!=null){
+			// url = url+"?"+payload;
+			// }
+			HttpGet httpGet = new HttpGet(url);
+			HttpGet httpGetWithHeaders = (HttpGet) buildHeaders(httpGet, headers, httpMethod);
+			Log.d(TAG, httpGetWithHeaders.toString() + " GET");
+			try {
+				HttpResponse response = httpclient.execute(httpGetWithHeaders);
+				responseParams.put("response", getResponseBody(response));
+				responseParams.put("status",
+				                   String.valueOf(response.getStatusLine().getStatusCode()));
+				return responseParams;
+			} catch (ClientProtocolException e) {
+				Log.d(TAG, "ClientProtocolException :" + e.toString());
+				return null;
```

Comment:
Is there any specific reason why we need to catch exceptions and return a value? Can't we throw a proper wrapped exception upwards here in this particular instance?